src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -174,14 +172,14 @@ public void MultpleTimers_PeriodicTimerIsntBlockedByABlockedTimer()
         {
             using (var t1 = new Timer((object obj) => are1.Set(), null, 1, -1))
             {
-                Assert.True(are1.WaitOne(250), "Should have received a second callback while blocked");
+                Assert.True(are1.WaitOne(1000), "Should have received a second callback while blocked");
                 are2.Set();
             }
         });
 
         using (var t1 = new Timer(tc, null, 1, -1))
         {
-            Assert.True(are2.WaitOne(500), "Blocking callback prevented a second timer from firing");
+            Assert.True(are2.WaitOne(2000), "Blocking callback prevented a second timer from firing");
```

Comment:
I just realized this test's name doesn't match what it's testing.  I'm also still a bit concerned about potential flakiness.  Here's a suggestion for how to rewrite it to hopefully be more reliable (pseudo-code):
```C#
[Fact]
public void MultipleTimers_PeriodTimerIsntBlocked()
{
    const int FailureTimeout = 10 * 1000;
    var mres = new ManualResetEventSlim();
    int callbacks = 2;
    var b = new Barrier(callbacks + 1);

    Timer t = null;
    t = new Timer(_ =>
    {
        if (Interlocked.Decrement(ref callbacks) >= 0)
        {
            Assert.True(b.SignalAndWait(FailureTimeout));
        }
        t.Dispose();
    }, null, 1, 50);
    Assert.True(b.SignalAndWait(FailureTimeout));
}
```